### PR TITLE
Add new cases of nat 

### PR DESCRIPTION
**Pull request by chloerh**
- nat for ipv6 is supported
- Start vm with 2 interfaces connected to 2 different nat network
and check net-dhcp-leases

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
# avocado run --vt-type libvirt virtual_network.iface_network.net_forward.net_nat.nat_ipv6
JOB ID     : d639a4e16516335faed9e05876c1f679ee91fc66
JOB LOG    : /root/avocado/job-results/job-2020-10-21T15.45-d639a4e/job.log
 (1/1) type_specific.local.virtual_network.iface_network.net_forward.net_nat.nat_ipv6: PASS (51.34 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 52.32 s

**Comment by chloerh** [comment]
Depend on https://github.com/avocado-framework/avocado-vt/pull/2788

**Comment by dzhengfy** [comment]
@yalzhang Could you help review from your POV?

**Review by yalzhang — APPROVED** [review]

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 1]
```diff
@@ -71,7 +73,7 @@ def prepare_pxe_boot():
         with open(pxe_file, 'w') as p_file:
             p_file.write(boot_txt)
 
-    def modify_iface_xml(sync=True):
+    def modify_iface_xml(params, sync=True):
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_network.py:76]: seems not used, so why adding it?

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 2]
```diff
@@ -698,6 +712,15 @@ def run_guest_libvirt(session):
                     net_forward.update({"dev": net_ifs[0]})
                     netxml.forward = net_forward
             logging.info("netxml before define is %s", netxml)
+            if with_2net:
+                net2_net_name = params.get('net2_net_name', 'net2')
+                net2_params = {k.replace('net2_', ''): v for k, v in params.items()
+                               if k.startswith('net2_')}
+                logging.debug(net2_params)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_network.py:719]: Please give more clear debug info, thanks

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 3]
```diff
@@ -724,11 +747,20 @@ def run_guest_libvirt(session):
         if change_iface_option:
             try:
                 if update_device:
-                    updated_iface = modify_iface_xml(sync=False)
+                    updated_iface = modify_iface_xml(params, sync=False)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_network.py:750]: as above

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 4]
```diff
@@ -724,11 +747,20 @@ def run_guest_libvirt(session):
...
                     virsh.update_device(vm_name, updated_iface.xml,
                                         ignore_status=False, debug=True)
                 else:
-                    modify_iface_xml()
+                    modify_iface_xml(params)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_network.py:754]: as above

File: libvirt/tests/src/virtual_network/iface_network.py [hunk 5]
```diff
@@ -751,6 +783,32 @@ def run_guest_libvirt(session):
...
+            for i in range(test_rounds):
+                logging.debug('Test round %d', i + 1)
+                os.remove(status_path % net1_br)
+                os.remove(status_path % net2_br)
+                virsh.net_destroy(net_name, **args)
+                virsh.net_destroy(net2_net_name, **args)
+                virsh.net_start(net_name, **args)
+                virsh.net_start(net2_net_name, **args)
+                virsh.start(vm_name, **args)
+                time.sleep(40)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_network.py:804]: why sleep so long here? Or can be replaced by wait_for?

Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_network.py:804]: I write it in the manual case. I have tested and found maybe we can replace it with 5s.

Comment (chloerh) [reply libvirt/tests/src/virtual_network/iface_network.py:804]: 5s seems reasonable

**Review by kylazhang — APPROVED** [review]

**Pull request merged**